brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx

```diff
@@ -1,39 +1,45 @@
 {% extends 'common.pyx' %}
 
-{% block template_support_code %}
-cdef int _start_idx = 0
-cdef double _current_time = 0.0
-{% endblock %}
-
 {% block maincode %}
 
-    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time } #}
-     
-    global _start_idx, _current_time
-    
-    # Check if we have been reset, if so reset the cache
-    if t<_current_time:
-        _start_idx = 0
-    
-    _current_time = t
+    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time, period, _lastindex } #}
 
     # TODO: We don't deal with more than one spike per neuron yet
-    cdef int _cpp_numspikes = 0
+    cdef int _cpp_numspikes   = 0
+    cdef float padding_before = t % period
+    cdef float padding_after  = (t + dt) % period
+    cdef float epsilon        = 1e-3*dt

```

Comment:
It's a bit of an opaque constant, I'm not sure this is something the user'd like to set. Is there any reason for not setting it to the smallest epsilon that still gives correct results for multiples of dt?